### PR TITLE
docs(contracts): add documentation for deployment workflow

### PR DESCRIPTION
**Pull request by 0xmad**
# Description

- [x] Add documentation for deploy steps
- [x] Add documentation for helpers
- [x] Add documentation for runners
- [x] Add some notes for task imports

## Additional Notes

N/A

## Related issue(s)

Related to #1096 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: contracts/tasks/helpers/ContractStorage.ts [hunk 1]
```diff
@@ -14,15 +19,33 @@ type TStorage = Record<
...
 
+  /**
+   * Initialize class properties only once
+   */
   private constructor() {
     this.db = low(new FileSync<TStorage>("./deployed-contracts.json"));
   }
 
+  /**
+   * Get signleton object
```
Comment (kittybest) [inline contracts/tasks/helpers/ContractStorage.ts:45]: typo: singleton

File: contracts/tasks/helpers/ContractStorage.ts [hunk 2]
```diff
@@ -72,28 +100,63 @@ export class ContractStorage {
       .write();
   }
 
+  /**
+   * Get contract instances from json file
+   *
+   * @param network - selected network
+   * @returns {[string, IStorageInstanceEntry][]} storage instance enties
```
Comment (kittybest) [inline contracts/tasks/helpers/ContractStorage.ts:107]: typo: entities

File: contracts/tasks/helpers/ContractStorage.ts [hunk 3]
```diff
@@ -72,28 +100,63 @@ export class ContractStorage {
...
+   * @param address - contract address
+   * @param network - selected network
+   * @returns {boolean} contract verified or not
+   */
   getVerified(address: string, network: string): boolean {
     return this.db.get(`${network}.verified.${address}`).value() as unknown as boolean;
   }
 
+  /**
+   * Set contract verification is json file
```
Comment (kittybest) [inline contracts/tasks/helpers/ContractStorage.ts]: typo: in

File: contracts/tasks/helpers/ContractStorage.ts [hunk 4]
```diff
@@ -72,28 +100,63 @@ export class ContractStorage {
...
   setVerified = (address: string, network: string, verified: boolean): void => {
     this.db.set(`${network}.verified.${address}`, verified).write();
   };
 
+  /**
+   * Get contract address by name from json file
+   *
+   * @param id - contract name
+   * @param network - selected network
+   * @returns {string | undefined} contract address
```
Comment (ctrlc03) [inline contracts/tasks/helpers/ContractStorage.ts:143]: don't think we need to add the type here?

File: contracts/tasks/helpers/ContractStorage.ts [hunk 5]
```diff
@@ -72,28 +100,63 @@ export class ContractStorage {
...
     return namedEntry?.address;
   }
 
+  /**
+   * Get contract address by name from json file
+   *
+   * @param id - contract name
+   * @param network - selected network
+   * @throws {Error} if there is no address the error will be thrown
+   * @returns {string} contract address
```
Comment (ctrlc03) [inline contracts/tasks/helpers/ContractStorage.ts:158]: same as above

**Review by ctrlc03 — APPROVED** [review]
Thank you, just some minor comments

File: contracts/tasks/helpers/ContractStorage.ts [hunk 6]
```diff
@@ -104,6 +167,13 @@ export class ContractStorage {
     return address;
   }
 
+  /**
+   * Get contract from json file with sizes and multi count
+   *
+   * @param deployer - deployer address
+   * @param network - selected network
+   * @returns {[entries: Map<string, string>, lenght: number, multiCount: number]}
```
Comment (kittybest) [inline contracts/tasks/helpers/ContractStorage.ts:175]: typo: length

File: contracts/tasks/helpers/Deployment.ts [hunk 7]
```diff
@@ -5,18 +5,38 @@ import type { EContracts, IDeployParams, IDeployStep, IDeployStepCatalog } from
...
+  /**
+   * Initialize class properties only once
+   */
   private constructor(hre?: HardhatRuntimeEnvironment) {
     this.stepCatalog = new Map([["full", []]]);
     this.hre = hre;
   }
 
+  /**
+   * Get signleton object
```
Comment (kittybest) [inline contracts/tasks/helpers/Deployment.ts:36]: typo: singleton

**Pull request merged**